### PR TITLE
Merge 177 to develop - 15/06/2020

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Merge 177 to develop
| Type?         | improvement
| Category?     | ME
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | ~
| How to test?  | ~

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19755)

**NOTE:**

Pay close attention to conflicted files:

```
CONFLICT (content): Merge conflict in themes/classic/templates/catalog/_partials/product-cover-thumbnails.tpl
CONFLICT (content): Merge conflict in tests/UI/pages/BO/orders/view.js
CONFLICT (content): Merge conflict in tests/UI/pages/BO/catalog/products/add.js
CONFLICT (content): Merge conflict in tests/UI/pages/BO/BObasePage.js
CONFLICT (content): Merge conflict in tests/UI/package.json
CONFLICT (content): Merge conflict in tests/UI/package-lock.json
CONFLICT (content): Merge conflict in tests/UI/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/06_displayDiscountedPrice.js
CONFLICT (content): Merge conflict in tests/UI/campaigns/functional/BO/08_design/07_linkWidget/01_CreateAndCheckFooterLinkWidget.js
CONFLICT (content): Merge conflict in tests/Integration/Behaviour/Features/Scenario/Customer/customer_management.feature
CONFLICT (content): Merge conflict in src/Core/Domain/ValueObject/Money.php
CONFLICT (content): Merge conflict in mails/themes/classic/core/order_conf.html.twig
CONFLICT (content): Merge conflict in admin-dev/themes/new-theme/js/pages/order/view/order-view-page.js
CONFLICT (content): Merge conflict in admin-dev/themes/new-theme/js/fos_js_routes.json
```

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

**Comment by matks** [comment]
Saw some mail templates changed => does it need re-computation ?

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
Seems OK to me, only 1 worry: can you check `themes/classic/templates/catalog/_partials/product-cover-thumbnails.tpl` ? I believe it's the "product cover" topic where we did back-and-forth so better check we have not, again, modified the behavior

**Review by matks — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

File: tests/UI/pages/BO/catalog/products/add.js [hunk 1]
```diff
@@ -23,7 +23,9 @@ module.exports = class AddProduct extends BOBasePage {
     this.previewProductLink = 'a#product_form_preview_btn';
     this.productOnlineSwitch = '.product-footer div.switch-input';
     this.productOnlineTitle = 'h2.for-switch.online-title';
+    this.productShortDescriptionTab = '#tab_description_short a';
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/products/add.js]:
```suggestion
```

File: tests/UI/pages/BO/catalog/products/add.js [hunk 2]
```diff
@@ -23,7 +23,9 @@ module.exports = class AddProduct extends BOBasePage {
...
     this.productShortDescriptionIframe = '#form_step1_description_short_1_ifr';
+    this.productDescriptionTab = '#tab_description a';
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/products/add.js]:
```suggestion
```

File: tests/UI/pages/BO/catalog/products/add.js [hunk 3]
```diff
@@ -58,14 +58,38 @@ module.exports = class AddProduct extends BOBasePage {
...
+  }
+
   /**
    * Set Name, type of product, Reference, price ttc, description and short description
    * @param productData
    * @return {Promise<void>}
    */
   async setBasicSetting(productData) {
     await this.setValue(this.productNameInput, productData.name);
+    await this.page.click(this.productDescriptionTab);
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/products/add.js]:
```suggestion
```

File: tests/UI/pages/BO/catalog/products/add.js [hunk 4]
```diff
@@ -58,14 +58,38 @@ module.exports = class AddProduct extends BOBasePage {
...
     await this.setValueOnTinymceInput(this.productDescriptionIframe, productData.description);
+    await this.page.click(this.productShortDescriptionTab);
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/products/add.js]:
```suggestion
```

File: tests/UI/pages/BO/BObasePage.js [hunk 5]
```diff
@@ -168,24 +169,27 @@ module.exports = class BOBasePage extends CommonPage {
...
+
+      await Promise.all([
+        this.page.click(parentSelector),
+        this.waitForVisibleSelector(`${parentSelector}.open`),
+      ]);
     }
-    await this.waitForVisibleSelector(`${linkSelector}.link-active`);
+    await this.scrollTo(linkSelector);
+    await this.clickAndWaitForNavigation(linkSelector);
+    await this.waitForVisibleSelector(`${linkSelector}.-active`);
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/BObasePage.js]:
```suggestion
    await this.waitForVisibleSelector(`${linkSelector}.link-active`);
```

**Comment by jolelievre** [comment]
> Saw some mail templates changed => does it need re-computation ?

No it's ok it's in the classic theme, no MJML sources like in the modern theme (but it could be fixed in the 177 branch as well)

> Seems OK to me, only 1 worry: can you check `themes/classic/templates/catalog/_partials/product-cover-thumbnails.tpl` ? I believe it's the "product cover" topic where we did back-and-forth so better check we have not, again, modified the behavior

Yeah it's ok, I paid close attention to this one

**Review by boubkerbribri — COMMENTED** [review]
Small changes for UI folder conflicts, I think the rest is OK.

**Review by matks — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by matks** [comment]
Thank you @jolelievre

**Pull request merged**